### PR TITLE
(maint) Update gemspec to assume at least the rubygems version included in ruby 1.9.3

### PR DESCRIPTION
**Pull request by kylog**

### CODE REVIEW

File: .gemspec [hunk 1]
```diff
@@ -37,19 +37,7 @@ Gem::Specification.new do |s|
...
-    else
-      s.add_dependency(%q<facter>, [">= 1.7", "< 3"])
-      s.add_dependency(%q<hiera>, [">= 2.0", "< 4"])
-    end
-  else
-    s.add_dependency(%q<facter>, [">= 1.7", "< 3"])
-    s.add_dependency(%q<hiera>, [">= 2.0", "< 4"])
-  end
+  s.specification_version = 3
+  s.add_runtime_dependency(%q<facter>, [">= 1.7", "< 3"])
```
Comment (ferventcoder) [inline .gemspec]: Is this still an appropriate upper bound?

Comment (kylog) [inline .gemspec]: No, but that's unrelated to dropping support for older rubygems, so I was thinking that would be a separate PR.

**Comment by joshcooper** [comment]
```
$ irb
requiirb(main):001:0> require 'rubygems'
=> false
...
irb(main):005:0> require 'pp'
=> true
...
irb(main):013:0> pp Gem::Specification.instance_methods.grep /required/
[:required_ruby_version,
 :required_rubygems_version,
 :required_ruby_version=,
 :required_rubygems_version=]
=> [:required_ruby_version, :required_rubygems_version, :required_ruby_version=, :required_rubygems_version=]
irb(main):014:0> pp Gem::Specification.instance_methods.grep /specification_version/         [:specification_version, :specification_version=]
=> [:specification_version, :specification_version=]
irb(main):015:0> pp Gem::Specification.instance_methods.grep /add_runtime_dep/
[:add_runtime_dependency]
=> [:add_runtime_dependency]
